design/decisions/why-lint-package/Why-flake8.md

```diff
@@ -0,0 +1,123 @@
+---
+title: "Why Flake8?"
+description: "Overview and rationale for choosing Flake8 as a code
+linting tool."
+author: "Richard"
+date: 2023-11-27
+categories:
+---
+
+## Context and Problem Statement
+
+Flake8 is a powerful linting tool for Python, integrating various
+functionalities to ensure code quality. Its underlying tools include
+pyflakes for identifying issues like undefined names, pycodestyle for
+enforcing PEP 8 style guide conventions, and mccabe for detecting code
+complexity. This integration provides a comprehensive solution for code
+analysis and style enforcement in Python development.
+
+Flake8's suite of tools includes pyflakes for issues identification,
+pycodestyle for style guide enforcement, and mccabe for complexity
+detection. These tools collectively ensure thorough code analysis,
+essential in Python development.
+
+## Decision Drivers
```

Comment:
This section is about what is pushing us to consider these tools. E.g. "Code reviews shouldn't involve commenting on the style" and "How to be consistent with code style within the team".